core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -32,24 +32,19 @@
  */
 public final class SqlStatements implements JdbiConfig<SqlStatements> {
 
-    private final Map<String, Object> attributes;
-    private TemplateEngine templateEngine;
-    private SqlParser sqlParser;
-    private SqlLogger sqlLogger;
-    private Integer queryTimeout;
-    private boolean allowUnusedBindings;
+    // cannot be a ConcurrentHashMap because of null keys
+    private final Map<String, Object> attributes = Collections.synchronizedMap(new HashMap<>());
+    private TemplateEngine templateEngine = new DefinedAttributeTemplateEngine();
+    private SqlParser sqlParser = new ColonPrefixSqlParser();
+    private SqlLogger sqlLogger = SqlLogger.NOP_SQL_LOGGER;
+    private Integer queryTimeout = null;
+    private boolean allowUnusedBindings = false;
```

Comment:
Initializing these fields here creates a bunch of garbage objects that get immediately clobbered when the copy constructor is used.

I actually prefer the symmetry of having statements in the default and copy constructors that you can compare. Makes it easy to see when you've forgotten to copy a field in the copy constructor.

Reply:
Agreeable. Compactness vs constructor symmetry, either way works for me.